system/core/document.php

```diff
@@ -49,7 +49,15 @@ public function __construct($file = false, $base_url = false) {
 				$contents = $file->read();
 			else
 				$contents = '<html><head></head><body></body></html>';
+			$previousSetting = libxml_use_internal_errors(true);
 			$this->loadHTML('<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">'.$contents);
+			$map = [LIBXML_ERR_WARNING => E_PARSE, LIBXML_ERR_ERROR => E_WARNING, LIBXML_ERR_FATAL => E_ERROR];
+			foreach (libxml_get_errors() as $error) {
+				if (801 !== $error->code) {
+					error_log($error->message, $map[$error->level]);
+				}
+			}
+			libxml_use_internal_errors($previousSetting);
```

Comment:
This seems unrelated to styling?

Reply:
The `$file` can contain non-strict html that would cause errors, therefore those errors are ignored. In my opinion the non-strictness is only for styling purposes.

Reply:
I've moved the libxml change to its own commit, and clarified in the code that this only ignores unknown tag errors, not *all* errors as I previously thought when quickly reading the code.